Invalid image format fix

bpisano opened the pull request:
## Summary
This pull request improve error handling when one of the images is not in a valid format, fixing issue #214.

## Details
Instead of logging the previous `err` variable as a string, I first check if it is possible to log the image name in the renamed `event` object. This way, we can better understand witch image is causing an issue. Previously, if we passed an invalid image format like `data:,`, resemble would log the `event` object as a string resulting in a strange `[event Object]` error. If the error still remain unknown, the message is now more explicit.

## How to test
You can simply invoke resemble with some invalid images url like so:
```js
resemble(validImageUrl)
  .compareTo(invalidImageUrl)
  .onComplete((data) => {
      console.log(data.error) // should now display a proper error message.
  });
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rsmbl/Resemble.js?pullRequest=216) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rsmbl/Resemble.js?pullRequest=216) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rsmbl/Resemble.js?pullRequest=216) it.</sub>

bpisano commented:
I fixed the code format for the CI, fixed the tests and added one for invalid image format. I also upgraded `canvas` to 2.9.0 (was 2.8.0) witch do not include breaking changes but improve the build for M1 Mac.
